src/commands/force/user/password/generate.ts

```diff
@@ -4,128 +4,66 @@
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-import * as os from 'os';
-import { flags, FlagsConfig, SfdxCommand } from '@salesforce/command';
-import { AuthInfo, Connection, StateAggregator, Messages, Org, SfError, User } from '@salesforce/core';
-import { PasswordConditions } from '@salesforce/core/lib/org/user';
-import { asNumber } from '@salesforce/ts-types';
+import {
+  arrayWithDeprecation,
+  Flags,
+  loglevel,
+  orgApiVersionFlagWithDeprecations,
+  requiredHubFlagWithDeprecations,
+} from '@salesforce/sf-plugins-core';
+import { Messages } from '@salesforce/core';
+import { ensureArray } from '@salesforce/kit';
+import { GenerateResult, UserPasswordGenerateBaseCommand } from '../../../../baseCommands/user/password/generate';
+
 Messages.importMessagesDirectory(__dirname);
 const messages = Messages.loadMessages('@salesforce/plugin-user', 'password.generate');
 
-interface PasswordData {
-  username?: string;
-  password: string;
-}
-
-export class UserPasswordGenerateCommand extends SfdxCommand {
+export class ForceUserPasswordGenerateCommand extends UserPasswordGenerateBaseCommand {
+  public static readonly summary = messages.getMessage('summary');
   public static readonly description = messages.getMessage('description');
-  public static readonly examples = messages.getMessage('examples').split(os.EOL);
-  public static readonly requiresUsername = true;
-  public static readonly supportsDevhubUsername = true;
-  public static readonly flagsConfig: FlagsConfig = {
-    onbehalfof: flags.array({
+  public static readonly examples = messages.getMessages('examples');
+  public static readonly flags = {
+    'on-behalf-of': arrayWithDeprecation({
+      aliases: ['onbehalfof'],
       char: 'o',
       description: messages.getMessage('flags.onBehalfOf'),
     }),
-    length: flags.integer({
+    length: Flags.integer({
       char: 'l',
       description: messages.getMessage('flags.length'),
       min: 8,
       max: 1000,
       default: 13,
     }),
     // the higher the value, the stronger the password
-    complexity: flags.integer({
+    complexity: Flags.integer({
       char: 'c',
       description: messages.getMessage('flags.complexity'),
       min: 0,
       max: 5,
       default: 5,
     }),
+    'target-dev-hub': { ...requiredHubFlagWithDeprecations, required: false, hidden: true },
```

Comment:
QA notes: I tried to steal this technique and it still acts required (throw when there's not a devhub set).  We wouldn't have the no-dev-hub scenario happen on NUTs so this got by us (org:create UT kinda found it, since the sbox commands don't require a hub).

am gonna add a non-required hub

Reply:
@mshanemc I did not run into this. I do not have a dev-hub config'd and the command works fine w/o the flag

```users user:list -o =========redacted==== 
Warning: The --target-dev-hub flag is deprecated and is not being used in this command. The flag will be removed in v57 or later.
=== Users in org 00DDP000005yeEr2AI

 Default Alias Username                      Profile Name         User Id            
 ─────── ───── ───────────────────────────── ──────────────────── ────────────────── 
 (A)           =========redacted==== System Administrator 005Dxxxxxxxxxxxxxxxx
```

Reply:
but I like the idea of a specialized flag to handle optional dev hub